Marshal/Marshal.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  M A R S H A L
+//
+//       ()
+//       /\
+//  ()--'  '--()
+//    `.    .'
+//     / .. \
+//    ()'  '()
+//
+//
+
+
+import Foundation
+
+
+public protocol Marshal {
+    subscript(key: KeyType) -> Any? { get }
+    func anyForKey(key: KeyType) throws -> Any
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A?
```

Comment:
`valueForKey` is needed because that is how you extract values if you are using the protocol as a data type. `anyForKey` is there in case somebody want to override the default behavior (i.e change the key separator or whatever).